llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp

```diff
@@ -4161,7 +4161,7 @@ struct MemorySanitizerVisitor : public InstVisitor<MemorySanitizerVisitor> {
 
   // Instrument AVX permutation intrinsic.
   // We apply the same permutation (argument index 1) to the shadow.
-  void handleAVXVpermilvar(IntrinsicInst &I) {
+  void handleAVXPermutation(IntrinsicInst &I) {
```

Comment:
Since this function is being used for many more cases, please augment it with assertions that check whether the instruction is suitable for this handler:
- number of operands
- type and size of each operand
- relation of input and output operands